### ISSUE TITLE
[Feature Request] Add functionality to help sync one babonus with a canonical bab stored elsewhere

### ISSUE THREAD

**Issue by p4535992** [krbz999/babonus#279]
**Describe the bug**

I think you forgot to expoe the keepId parameter on the api `game.modules.get("babonus").api.embedBabonus`

**To Reproduce**

The method `game.modules.get("babonus").api.embedBabonus(object, bonus)` do not have the keepId parameter
The method `BabonusWorkshop._embedBabonus(object, bonus, keepId);` has a additonal parameter but is not exposed

**Expected behavior**

**Screenshots**

**Setup**

**Additional context**

The keeId can be useful on ` copyBonus` and `moveBonus` too

**Comment by krbz999** [comment]
This was a conscious decision since it is only used internally. Is there a use case that would necessitate exposing it to the API?

**Comment by p4535992** [comment]
It's a use case on a piece of code where I dynamically transfer bonuses from objects A and B to object C.
Basically, there's a step where I add the bonuses from object A to object C and a step where I add the bonuses from effect B to object C.

So at some point C has both A's and B's bonuses, now in case A and B had a Bonus with the same name to distinguish them I would have to do some additional code which I would avoid if I could use ids.

This is a problem in the next step where I would like to remove the effects of B from C and at a later time the effects of A from C.

Start C -> C
Add A -> C (A)
Add B -> C (A + B)
Remove A -> C (B)
Remove B  -> C

However if it is something you don't want to implement it doesn't matter **I will make sure I don't use bonus with the same name in more than one object**. simple enough.

**Comment by krbz999** [comment]
A better change may simply be to change the return value to be the babs created/deleted/moved.

**Comment by p4535992** [comment]
Can be useful , in other use cases, in that I don't think it solves the current problem because I still wouldn't have a "track" of which bonus came to me from object A and which from 'object B, 

I would need something like the "origin" field of active effects to have a reference to the object from which copied/moved/recovered the bonus.

**Comment by krbz999** [comment]
What exactly is the use case here that cannot be resolved by simply turning a bab on or off?

**Comment by krbz999** [comment]
I understand the *what* the first time around, but not the *why*, and the latter is what is gonna be the reason to consider a feature request.

**Comment by p4535992** [comment]
you must have answered me while, I was finishing reissuing the ticket:

> _What exactly is the use case here that cannot be resolved by simply turning a bab on or off?_
> 
> This is an automated solution , because the players "mess up" with babonus, so I decided not to let them play with it.
> 
> Fixing them by hand is tiring, so I was trying to automate the process as much as possible.

**Comment by krbz999** [comment]
An automated solution to what?
The process of what?

**Comment by p4535992** [comment]
The preconception is that players don't have access to the babonus ui, not being able as a GM to go and set every single item (all the players items are itemC) every time that for example I mistake the configuration of a bonus or the like I have to go by hand to fix them one by one, I decided therefore to set "basic" items (ItemA and ItemB) on which I set the correct bonuses gradually.

At this point I update the bonuses of the many ItemCs in the world using the new configurations of ItemA and ItemB, if by chance ItemA and ItemB have bonuses with the same name I have a hard time distinguishing them, if it is a use case that does not have to subsist, it is perfectly fine I will use different names (e.g. "+1 damage Itam A", +1 damage "item B") and end of story.
